web/src/components/workflow-editor/workflow-editor.tsx

```diff
@@ -96,89 +110,120 @@ export default function WorkflowEditor({
 		router,
 	]);
 
+	useEffect(() => {
+		if (!workflow || !encryptedSecrets) {
+			return;
+		}
+		const secretIds = new Set(encryptedSecrets.map((s) => s.secretId));
+		const missingSecrets = workflow?.nodes
+			.filter((node) => node.type === "action")
+			.reduce((acc, node) => {
+				const secretsForNode = Object.values(node.secretsMapping);
+				secretsForNode.forEach((secretId, secretIdx) => {
+					if (!secretIds.has(secretId as string)) {
+						if (!acc.has(node.id)) {
+							acc.set(node.id, new Set());
+						}
+						acc.get(node.id)?.add(secretIdx);
```

Comment:
I just noticed that the secrets mapping in `TEditorWorkflowActionNode` is of type `z.record(z.string(), z.string())` and not an array. So, the data structure should then be `Map<string, Set<string>>` while we add the `secretsMapping` to the set. So, you now need to iterate over the keys of `node.secretsMapping` instead of the values.